corehq/apps/registration/forms.py

```diff
@@ -487,24 +490,38 @@ class AdminInvitesUserForm(forms.Form):
     email = forms.EmailField(label="Email Address",
                              max_length=User._meta.get_field('email').max_length)
     role = forms.ChoiceField(choices=(), label="Project Role")
+    profile = forms.ChoiceField(choices=(), label="Profile", required=False)
 
     def __init__(self, data=None, excluded_emails=None, is_add_user=None, location=None,
                  role_choices=(), *, domain, **kwargs):
         super(AdminInvitesUserForm, self).__init__(data=data, **kwargs)
         domain_obj = Domain.get_by_name(domain)
         self.fields['role'].choices = role_choices
-        if domain_obj.commtrack_enabled:
+        show_profile = False
+        show_location = False
+        if domain_obj:
             self.fields['location_id'] = forms.CharField(label='Primary Location', required=False,
-                                                          widget=LocationSelectWidget(domain_obj.name),
-                                                          help_text=EMWF.location_search_help,
-                                                          initial=location.location_id if location else '')
-            self.fields['program'] = forms.ChoiceField(label="Program", choices=(), required=False)
-            programs = Program.by_domain(domain_obj.name)
-            choices = list((prog.get_id, prog.name) for prog in programs)
-            choices.insert(0, ('', ''))
-            self.fields['program'].choices = choices
+                                                         widget=LocationSelectWidget(domain_obj.name),
+                                                         help_text=EMWF.location_search_help,
+                                                         initial='')
+            show_location = True
+
+            if domain_has_privilege(domain_obj.name, privileges.APP_USER_PROFILES):
+                from corehq.apps.users.views.mobile import UserFieldsView
+                definition = CustomDataFieldsDefinition.get(domain_obj.name, UserFieldsView.field_type)
+                profile_choices = []
+                if definition:
+                    profiles = definition.get_profiles()
+                    profile_choices = [('', '')] + [(profile.id, profile.name) for profile in profiles]
+                self.fields['profile'].choices = profile_choices
+                show_profile = True
```

Comment:
nit: could we set `show_profile`' directly as `domain_has_privilege(domain_obj.name, privileges.APP_USER_PROFILES)` and use it for the conditional check? And something similar for `show_location`

Reply:
no, that wouldn't work because it lives within the `domain_obj` conditional check. But another small clean up was done [764d0e4](https://github.com/dimagi/commcare-hq/pull/34466/commits/764d0e4a5145a20f98acb3e7ab1263e33bcc26f4)